tests/cpp/backends/amdgpu_device_test.cpp

```diff
@@ -0,0 +1,102 @@
+#include "gtest/gtest.h"
+
+#ifdef TI_WITH_AMDGPU
+#include "taichi/ir/ir_builder.h"
+#include "taichi/rhi/amdgpu/amdgpu_driver.h"
+#include "taichi/rhi/amdgpu/amdgpu_context.h"
+#include "taichi/rhi/amdgpu/amdgpu_device.h"
+#include "tests/cpp/program/test_program.h"
+
+namespace taichi {
+namespace lang {
+TEST(AMDGPU, CreateDeviceAndAlloc) {
+  std::unique_ptr<amdgpu::AmdgpuDevice> device =
+      std::make_unique<amdgpu::AmdgpuDevice>();
+  EXPECT_TRUE(device != nullptr);
+  taichi::lang::Device::AllocParams params;
+  params.size = 1048576;
+  params.host_read = false;
+  params.host_write = false;
+  const taichi::lang::DeviceAllocation device_alloc =
+      device->allocate_memory(params);
+
+  // The purpose of the device_alloc_guard is to rule out double free
+  const taichi::lang::DeviceAllocationGuard device_alloc_guard(device_alloc);
```

Comment:
nit: would you mind elaborating when the double free happens? Thanks!

Reply:
This is to prevent manual dealloc_memory, but of course there is no double free problem here
I write this UT with reference to DX12, but I found redundancy in this UT and [this pr](https://github.com/taichi-dev/taichi/pull/6552)

Reply:
Yes, this is from the DX12 test, and I didn't find any double free when migrating this test to vulkan and opengl.